src/test/java/de/unijena/cheminf/mortar/model/fragmentation/algorithm/AlkylStructureFragmenterTest.java

```diff
@@ -0,0 +1,754 @@
+        for (IAtomContainer tmpAC : anACSet.atomContainers()) {
+            tmpSmilesList.add(ChemUtil.createUniqueSmiles(tmpAC, false));
+        }
+        return tmpSmilesList;
+    }
+    /**
+     * Utility method to generate SMILES notation strings for a given List containing IAtomContainer instances.
+     *
+     * @param anACList given List of IAtomContainers
+     * @return List with SMILES Strings of originally given List of IAtomContainers
+     */
+    private List<String> generateSMILESFromACList(List<IAtomContainer> anACList) {
+        List<String> tmpReturnSmilesList = new ArrayList<>(anACList.size());
+        for (IAtomContainer tmpAC: anACList) {
+            tmpReturnSmilesList.add(ChemUtil.createUniqueSmiles(tmpAC, false));
+        }
+        return tmpReturnSmilesList;
+    }
+    /**
+     * Private method to read a given structure file to a CDK atomcontainer set.
+     *
+     * @param aFileName Name of the file to read from
+     * @return IAtomContainerSet with the read structures as AtomContainers
+     * @throws FileNotFoundException if no file with the given name can be located
+     * @throws URISyntaxException if given name of file cannot be parsed as URI reference
+     */
+    private IAtomContainerSet readStructuresToACSet(String aFileName) throws FileNotFoundException, URISyntaxException {
+        URL tmpURL = this.getClass().getResource("/" + aFileName);
+        File tmpResourceFile = Paths.get(Objects.requireNonNull(tmpURL).toURI()).toFile();
+        IteratingSDFReader tmpSDFReader = new IteratingSDFReader(new FileReader(tmpResourceFile), new SilentChemObjectBuilder());
+        AtomContainerSet tmpACSet = new AtomContainerSet();
+        String tmpIndexString = "ASFTest.AtomContainerIndex";
+        int tmpIndex = 0;
+        while (tmpSDFReader.hasNext()) {
+            IAtomContainer tmpAtomContainer = tmpSDFReader.next();
+            tmpAtomContainer.setProperty(tmpIndexString, tmpIndex);
+            try {
+                ChemUtil.saturateWithHydrogen(tmpAtomContainer);
+            } catch (CDKException aCDKException) {
+                throw new RuntimeException(aCDKException);
+            }
+            tmpACSet.addAtomContainer(tmpAtomContainer);
+            tmpIndex++;
+        }
+        return tmpACSet;
+    }
+    /**
+     * Private method returning a local instance of AlkylStructureFragmenter with default settings for a given molecule.
+     *
+     * @param aMolecule given molecule to be assessed
+     * @param aShouldBeFilteredStatement boolean value to set pre-fragmentation task
+     * @param aShouldBePreprocessedStatement boolean value to set pre-fragmentation task
+     * @param aCanBeFragmentedStatement boolean value to set pre-fragmentation task
+     * @return AlkylStructureFragmenter instance with default settings
+     */
+    private AlkylStructureFragmenter getDefaultASFInstance(IAtomContainer aMolecule, boolean aShouldBeFilteredStatement,
```

Comment:
as stated above, instantiating a fragmenter instance and testing the molecule with the preprocessing methods should be separate.